### PR TITLE
8338668: Test javax/swing/JFileChooser/8080628/bug8080628.java doesn't test for GTK L&F

### PR DESCRIPTION
**Pull request by kumarabhi006**
This test should have failed with the [changeset](https://github.com/openjdk/jdk/pull/20612) proposed for [JDK-4795384](https://bugs.openjdk.org/browse/JDK-4795384) but it didn't because GTK look and feel isn't supported in headless environment.

Fix is to run the test with headful environment on linux platform. 

Tested in CI pipeline and link attached to JBS.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338668](https://bugs.openjdk.org/browse/JDK-8338668): Test javax/swing/JFileChooser/8080628/bug8080628.java doesn't test for GTK L&amp;F (**Bug** - P4)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20671/head:pull/20671` \
`$ git checkout pull/20671`

Update a local copy of the PR: \
`$ git checkout pull/20671` \
`$ git pull https://git.openjdk.org/jdk.git pull/20671/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20671`

View PR using the GUI difftool: \
`$ git pr show -t 20671`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20671.diff">https://git.openjdk.org/jdk/pull/20671.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20671#issuecomment-2303816250)

### CODE REVIEW

File: test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java [hunk 1]
```diff
@@ -33,7 +33,17 @@
...
+ * @modules java.desktop/sun.swing
+ * @run main bug8080628
+ */
+
+/*
+ * @test
+ * @bug 8080628
+ * @key headful
+ * @summary No mnemonics on Open and Save buttons in JFileChooser.
+ * @requires os.family == "linux"
```
Comment (alisenchung) [inline test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java:46]: Looks like jtreg tags were accidentally duped. Did you mean to only replace the author tag with the requires tag?

File: test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java
Comment (alisenchung) [inline test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java:1]: Could you also clean up the main method and remove the extra overhead in the SwingUtilities.invokeAndWait call? Also remember to update copyright years

File: test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java (hunk 1 above)
Comment (honkar-jdk) [reply test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java:46]: @alisenchung Multiple jtreg headers are added to run with different configuration
The 1st one runs in headless mode on platforms other than linux  `@requires os.family != "linux"`
The 2nd one runs in headful mode on linux `@requires os.family == "linux"`

**Comment by honkar-jdk** [comment]
@kumarabhi006 
> Fix is to run the test with headful environment on linux platform.

Does this have to do with not loading the gtk files in headless mode? 
https://github.com/openjdk/jdk/blob/master/src/java.desktop/unix/native/libawt_xawt/awt/gtk_interface.c#L26C1-L28C7

File: test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java (hunk 1 above)
Comment (kumarabhi006) [reply test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java:46]: @alisenchung As @honkar-jdk explained this is to run different configuration for different platforms.

File: test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java [hunk 2]
```diff
@@ -102,5 +114,4 @@ private static void runTest() {
             exception = e;
         }
     }
-
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java:105]: I'd rather keep this blank line. Most classes have blank line between the closing brace of the last method and the closing brace of the class itself.

**Review by aivanov-jdk — APPROVED** [review]
Please, update the copyright year.

**Comment by aivanov-jdk** [comment]
> @kumarabhi006
> 
> > Fix is to run the test with headful environment on linux platform.
> 
> Does this have to do with not loading the gtk files in headless mode? https://github.com/openjdk/jdk/blob/master/src/java.desktop/unix/native/libawt_xawt/awt/gtk_interface.c#L26C1-L28C7

Yes, that's exactly the problem.

Abhishek and I discussed !20612, and I expected this test to fail with the changes on the review but it didn't.

File: test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java
Comment (kumarabhi006) [reply test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java:1]: I think that is ok.. it's pretty simple where runnable interface run method is overridden and is supposed to run the test.
copyright year updated.

**Comment by kumarabhi006** [comment]
@aivanov-jdk Please re-review again. Due to recent changes, it is required to review the latest commit.

**Review by aivanov-jdk — APPROVED** [review]

File: test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java
Comment (aivanov-jdk) [reply test/jdk/javax/swing/JFileChooser/8080628/bug8080628.java:1]: What do you mean? Replacing `new Runnable` with lambda?

Does it add value? Sure the code becomes shorter… However, the test still does its job. And IDE still shows the compact version displaying the body of the `run` method only.

Given that no other code is changed, except for adding additional output, I'd rather keep it as it is.

**Comment by honkar-jdk** [comment]
Test fix looks good. Works as expected on linux and non-linux platforms.

**Review by honkar-jdk — APPROVED** [review]

**Review by prrace — APPROVED** [review]
I guess this is OK. It is a bit of an anomaly that GTK requires headful and not so easy to explain to users, but it is a lot easier to fix a test than deal with making GTK work in headless mode ..

**Pull request closed without merging**